src/lightning_lite/utilities/data.py

```diff
@@ -0,0 +1,425 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import inspect
+import os
+from collections import OrderedDict
+from contextlib import contextmanager
+from functools import partial
+from typing import Any, Callable, Dict, Generator, Iterable, Optional, Set, Tuple, Type, Union
+
+from torch.utils.data import BatchSampler, DataLoader, IterableDataset, Sampler
+
+from lightning_lite.utilities.enums import LightningEnum
+from lightning_lite.utilities.exceptions import MisconfigurationException
+from lightning_lite.utilities.rank_zero import rank_zero_warn
+from lightning_lite.utilities.seed import pl_worker_init_function
+
+
+class _WrapAttrTag(LightningEnum):
```

Comment:
Shall this be StrEnum from utils?

Reply:
Yes, it will happen here  #14558

Reply:
Yes, but https://github.com/Lightning-AI/lightning/pull/14558 needs to land first